ceph.spec.in

```diff
@@ -109,9 +109,11 @@
 
 # disable dwz which compresses the debuginfo
 %global _find_debuginfo_dwz_opts %{nil}
+%if ! 0%{?suse_version}
 # use multi-threaded xz compression: xz level 7 using ncpus threads
 %global _source_payload w7T%{_smp_build_ncpus}.xzdio
 %global _binary_payload w7T%{_smp_build_ncpus}.xzdio
+%endif
```

Comment:
is `%{jobs}` small enough to avoid the OOM?

Reply:
On SUSE, `%jobs` is set in the `%build` phase by this: https://github.com/ceph/ceph/blob/master/ceph.spec.in#L1214-L1216

So it's not available here.

I think enabling the parallel compression in OBS is a worthy goal, since it might speed up the builds significantly. I tried a little and was not able to get it to work. Hence this PR which simply reverts to the previous (working) state.

Reply:
> So `%jobs` is not available here.

Or, if it is available, it is set to the maximum number of threads the CPUs can sustain (in the OBS, typically 12 or 16),  and this causes the build to OOM every time.

But I have not tried changing `%_smp_build_ncpus` to `%jobs` in this code. It might work. I will try it and let you know.

Reply:
okay. thought we could move `%limit_build -m 2600` before this line. but anyway.

Reply:
The problem in the OBS is that the code which sets a sane `%jobs` value is run in the `%build` phase and I was not able to get it to run early enough to be of use here.

Reply:
> But I have not tried changing `%_smp_build_ncpus` to `%jobs` in this code. It might work. I will try it and let you know.

yes. that's what i meant to suggest: if it's small enough, could just use it to replace `_smp_build_ncpus` everywhere.

Reply:
> okay. thought we could move `%limit_build -m 2600` before this line. but anyway.

I had the same idea. Alas, moving that line up before this line causes the build to fail.

Reply:
One possibility would be to copy the `%limit_build` macro definition into the spec file. I think it fails because the macro definition is being brought in via BuildRequires.

Reply:
Anyway, I'd rather tinker with speed optimization _after_ the OBS master build is restored to working order.